crates/hir-analysis/src/ty/diagnostics.rs

```diff
@@ -581,6 +731,10 @@ impl TraitConstraintDiag {
             Self::ConcreteTypeBound(_, _) => {
                 "trait bound for concrete type is not allowed".to_string()
             }
+
+            Self::DependentTyBound(_, _) => {
+                "trait bound for dependent type is not allowed".to_string()
```

Comment:
Yeah, it's confusing. I'll fix it tomorrow.
I'll also remove all `dependent` usage from the code base since we decided to postpone designing a dependent type system for now.

Reply:
Done.